engines/scumm/macgui/macgui_v6.cpp

```diff
@@ -0,0 +1,1059 @@
+
+	if (_vm->_game.id != GID_MANIAC) {
+		window->addSubstitution("");
+		window->addSubstitution("");
+		window->addSubstitution("");
+		window->addSubstitution(_gameName);
+
+		interactionPopUp = (MacPopUpMenu *)window->getWidget(kWidgetPopUpMenu, 0);
+		videoQualityPopUp = (MacPopUpMenu *)window->getWidget(kWidgetPopUpMenu, 1);
+	} else {
+		videoQualityPopUp = (MacPopUpMenu *)window->getWidget(kWidgetPopUpMenu, 0);
+	}
+
+	// Note: The video quality menu contains an additional "Graphics
+	// Smoothing" entry. I don't know why it doesn't show up in the
+	// original, but as long as we disabled the pop-up that's not a
+	// problem. My future self can thank me later.
+
+	videoQualityPopUp->setValue(1);
+	videoQualityPopUp->setEnabled(false);
+
+	window->setDefaultWidget(buttonOk);
+
+	if (_vm->_game.id == GID_TENTACLE) {
+		// Yes, the frames really are supposed to be slightly
+		// misaligned to match the original appearance.
+
+		drawDottedFrame(window, Common::Rect(12, 41, 337, 113), 21, 137);
+		drawDottedFrame(window, Common::Rect(11, 130, 336, 203), 20, 168);
+
+		sliderMusicVolume = addSlider(window, 152, 63, 147, 17);
+		voiceVolumeSlider = addSlider(window, 152, 87, 147, 17);
+		sliderTextSpeed = addSlider(window, 151, 177, 147, 9);
+	} else if (_vm->_game.id == GID_MANIAC) {
+		sliderMusicVolume = addSlider(window, 152, 41, 147, 17);
+		sliderTextSpeed = addSlider(window, 152, 72, 147, 10, 5);
+	} else if (_vm->_game.id == GID_SAMNMAX || _vm->_game.id == GID_DIG) {
+		drawDottedFrame(window, Common::Rect(12, 41, 337, 136), 21, 137);
+		drawDottedFrame(window, Common::Rect(12, 156, 337, 229), 20, 168);
+
+		sliderMusicVolume = addSlider(window, 152, 63, 147, 17);
+		effectVolumeSlider = addSlider(window, 152, 87, 147, 17);
+		voiceVolumeSlider = addSlider(window, 152, 111, 147, 17);
+		sliderTextSpeed = addSlider(window, 152, 203, 147, 9);
+	} else if (_vm->_game.id == GID_FT) {
+		drawDottedFrame(window, Common::Rect(12, 41, 337, 164), 21, 137);
+		drawDottedFrame(window, Common::Rect(12, 184, 337, 257), 20, 168);
+
+		sliderMusicVolume = addSlider(window, 152, 63, 147, 17);
+		effectVolumeSlider = addSlider(window, 152, 87, 147, 17);
+		voiceVolumeSlider = addSlider(window, 152, 111, 147, 17);
+		sliderTextSpeed = addSlider(window, 152, 231, 147, 9);
+
+		spoolMusicCheckbox = (MacCheckbox *)window->getWidget(kWidgetCheckbox, 0);
+	}
+
+	if (interactionPopUp) {
+		switch (_vm->_voiceMode) {
+		case 0: // Voice Only
+			interactionPopUp->setValue(1);
+			break;
+		case 1: // Voice And Text
+			interactionPopUp->setValue(2);
+			break;
+		case 2: // Text Only
+			interactionPopUp->setValue(0);
+			break;
+		default:
+			warning("MacGuiImpl::MacV6Gui::runOptionsDialog(): Invalid voice mode %d", _vm->_voiceMode);
+			break;
+		}
+	}
+
+	int musicVolume = 0;
+	int effectVolume = 0;
+	int voiceVolume = 0;
+
+	if (sliderMusicVolume) {
+		musicVolume = _vm->_mixer->getVolumeForSoundType(Audio::Mixer::kMusicSoundType) / 16;
```

Comment:
It isn't possible to read imuse (or any other non-digital) sound volume from the mixer. You have to use ConfMan.getInt("music_volume").